### PR TITLE
Implementing likes and comments

### PR DESCRIPTION
**Pull request by edthomas93**

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -8,24 +8,45 @@
   Home
 <% end %>
 
-<% @posts.reverse_each do |post| %>
+<% @posts.order(:created_at).reverse_each do |post| %>
```
Comment (pliantmeerkat) [inline app/views/posts/index.html.erb:11]: Good job!

File: db/schema.rb [hunk 2]
```diff
@@ -10,11 +10,20 @@
...
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20180830111808) do
+ActiveRecord::Schema.define(version: 20180905085619) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "comments", force: :cascade do |t|
+    t.string "username"
```
Comment (pliantmeerkat) [inline db/schema.rb]: should be not NULL: t.string 'username', null: false
also no double quotes

File: db/schema.rb [hunk 3]
```diff
@@ -10,11 +10,20 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20180830111808) do
+ActiveRecord::Schema.define(version: 20180905085619) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "comments", force: :cascade do |t|
```
Comment (pliantmeerkat) [inline db/schema.rb:18]: get rid of double quotes

File: spec/controllers/comments_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe CommentsController, type: :controller do
```
Comment (pliantmeerkat) [inline spec/controllers/comments_controller_spec.rb:3]: do we need this, if not take it out

File: spec/features/feature_liking_posts_spec.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+RSpec.feature 'Post likes', type: :feature do
+  before(:each) { visit('/') }
```
Comment (pliantmeerkat) [inline spec/features/feature_liking_posts_spec.rb:3]: refactor this down, add a heper method calledlogin and post maybe?

File: spec/features/feature_liking_posts_spec.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  scenario 'user can like posts' do
+    add_test_data_to_db
+    login_to_acebook
+    click_link("View Posts")
+    id = Post.where(message: 'hello i am a test message').first.id
+    save_and_open_page
```
Comment (pliantmeerkat) [inline spec/features/feature_liking_posts_spec.rb]: remove debugging commands

File: spec/features/feature_post_comments_spec.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+#require '../app/models/post'
+RSpec.feature 'User can comment', type: :feature do
+  before(:each) { visit('/') }
+  scenario 'User can comment on posts' do
```
Comment (pliantmeerkat) [inline spec/features/feature_post_comments_spec.rb]: helper methods for this, spec should be < 5 lines ideally

File: spec/helpers/comments_helper_spec.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# the CommentsHelper. For example:
+#
+# describe CommentsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe CommentsHelper, type: :helper do
```
Comment (pliantmeerkat) [inline spec/helpers/comments_helper_spec.rb:13]: remove?

File: app/controllers/like_controller.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class LikeController < ApplicationController
+  def create
+    @post = Post.find(params["posts"["post"]]["id"].to_i)
```
Comment (pliantmeerkat) [inline app/controllers/like_controller.rb]: double quotes - should be single quotes

File: app/controllers/posts_controller.rb [hunk 10]
```diff
@@ -24,4 +28,10 @@ def post_params
   def getUser(post)
     return User.find(post.user_id)
   end
+
+  def like(post)
+    if !current_user.liked? @post
```
Comment (tblanchard01) [inline app/controllers/posts_controller.rb]: is this old code as per previous discussion? Please kindly delete if so.

File: app/views/posts/index.html.erb [hunk 11]
```diff
@@ -8,24 +8,45 @@
...
 
   <div class="post_actions">
-    <%= button_to 'Like', posts_path, method: :post %>
+    <%= button_to 'Like', "/like?post[id]=#{post.id}", id: 'like'+post.id.to_s, method: :post %>
   </div>
 
+  <h3> Comments </h3>
+
+  <% post.comments.each do |comment| %>
+    <div class="well">
```
Comment (pliantmeerkat) [inline app/views/posts/index.html.erb]: what is well class - make specific please

File: config/routes.rb [hunk 12]
```diff
@@ -1,7 +1,10 @@
 Rails.application.routes.draw do
   devise_for :users
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  resources :posts
+  resources :posts do
```
Comment (pliantmeerkat) [inline config/routes.rb:4]: 1 liner this possibly - :posts { ... }

File: db/migrate/20180905085619_create_comments.rb [hunk 13]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[5.1]
+  def change
+    create_table :comments do |t|
+      t.string :username
+      t.text :body
+      t.references :post, foreign_key: true
+
```
Comment (pliantmeerkat) [inline db/migrate/20180905085619_create_comments.rb]: delete this blank line

**Review by pliantmeerkat — CHANGES_REQUESTED** [review]
fix the essential issues, mostly styling

**Review by tblanchard01 — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 2 above)
Comment (velvetsnowman) [reply db/schema.rb]: we cant change schema, autogenerated

Comment (edthomas93) [reply db/schema.rb]: we did rails db:rollback, edited the migration file so username and body were `null: false`.
you should not use db:rollback if pushed to master (in prod)

File: spec/controllers/comments_controller_spec.rb (hunk 4 above)
Comment (edthomas93) [reply spec/controllers/comments_controller_spec.rb:3]: I think these are for when we implement unit tests on each of the functions,  so we'll need them eventually

File: db/schema.rb (hunk 3 above)
Comment (edthomas93) [reply db/schema.rb:18]: it's in the schema, so we'll leave it

File: spec/helpers/comments_helper_spec.rb (hunk 8 above)
Comment (edthomas93) [reply spec/helpers/comments_helper_spec.rb:13]: these are autogenerated in all the helper spec files. I think we should leave them in until we're done

File: config/routes.rb (hunk 12 above)
Comment (edthomas93) [reply config/routes.rb:4]: it didn't work, so leaving as is

**Review by pliantmeerkat — APPROVED** [review]

**Review by tblanchard01 — APPROVED** [review]

**Pull request merged**